server/webserver.go

```diff
@@ -32,27 +28,18 @@ func (s *webServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, "no body", http.StatusBadRequest)
 		return
 	}
-	jsonBytes, readErr := ioutil.ReadAll(r.Body)
-	r.Body.Close()
-	if readErr != nil {
-		http.Error(w, "failed to read incoming request", http.StatusBadRequest)
-		return
-	}
+
 	h := Handler(strings.TrimPrefix(r.URL.Path, s.path+"/"))
 	if h == HandlerGetRepo {
 		start := time.Now()
-		s.r.WriteRepoBytes(w)
 		w.Header().Set("Content-Type", "application/json")
+		s.r.WriteRepoBytes(w)
 		addMetrics(h, start, nil, nil, sourceWebserver)
 		return
 	}
-	reply, errReply := handleRequest(s.r, h, jsonBytes, "webserver")
-	if errReply != nil {
-		http.Error(w, errReply.Error(), http.StatusInternalServerError)
-		return
-	}
-	_, err := w.Write(reply)
-	if err != nil {
-		Log.Error("failed to write webServer reply", zap.Error(err))
+
+	if err := handleRequest(s.r, h, r.Body, w, "webserver"); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
 	}
+	r.Body.Close()
```

Comment:
You should check the error from calling .Close()

Reply:
Not really needed in this case but I will add it. 
Very important to check close errors when you work with files: https://www.joeshaw.org/dont-defer-close-on-writable-files/

Reply:
https://stackoverflow.com/questions/47293975/should-i-error-check-close-on-a-response-body

To quote:

> There are two things to consider: What would you do with it if you checked it and there was an error? And, what would the side-effects be if there was an error?
> In most cases, for closing a response body, the answer to both questions is... absolutely nothing. If there's nothing you'd do if there was an error and the error has no appreciable impact, there's no reason to check it.

Unlike files, where we indeed had a problem with not handling close.